### PR TITLE
WebGPU-Report errors in Promise or void returning operations

### PR DESCRIPTION
**Pull request by kunalmohan**
This also updates GPUBuffer mapping to match latest spec.

r?@kvark

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

### CODE REVIEW

File: components/webgpu/lib.rs [hunk 1]
```diff
@@ -67,6 +67,8 @@ pub enum WebGPURequest {
     BufferMapAsync {
         sender: IpcSender<WebGPUResponseResult>,
         buffer_id: id::BufferId,
+        device_id: id::DeviceId,
+        scope_id: Option<u64>,
```
Comment (kvark) [inline components/webgpu/lib.rs:71]: If we are sending the scope ID with every operation, maybe we should move it out of the num?
Also, consider changing it to `Option<NonZeroU64>` to reduce the size

**Review by kvark — APPROVED** [review]

**Review by kvark — APPROVED** [review]

**Comment by jdm** [comment]
@bors-servo retry
* network hiccough

**Comment by kunalmohan** [comment]
Looks like CI doesn't want any more of my PRs to merge :laughing:

**Comment by jdm** [comment]
@bors-servo retry
* #25522

**Pull request merged**